packages/legend-extension-dsl-data-space-studio/src/stores/DataSpaceEditorState.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { guaranteeType } from '@finos/legend-shared';
+import { action, computed, makeObservable, observable } from 'mobx';
+import {
+  type EditorStore,
+  ElementEditorState,
+} from '@finos/legend-application-studio';
+import type { PackageableElement } from '@finos/legend-graph';
+import { DataSpace } from '@finos/legend-extension-dsl-data-space/graph';
+
+export class DataSpaceEditorState extends ElementEditorState {
+  dataSpace: DataSpace;
+
+  constructor(editorStore: EditorStore, element: PackageableElement) {
+    super(editorStore, element);
+    this.dataSpace = guaranteeType(
```

Comment:
remove this and match how its done in `TextEditorState` with a get property instead but for DataSpace

  get textElement(): Text {
    return guaranteeType(
      this.element,
      Text,
      'Element inside text element editor state must be a text element',
    );
  }

Reply:
changed